### ISSUE TITLE
bad_alloc when trying to read from an empty lmdb

### ISSUE THREAD

**Issue by BenWiederhake** [BVLC/caffe#4124]
Hello,

tl;dr: caffe (current master) crashes when one of the lmdb's is empty.

Using the imagenet example, with few modifications:  when creating the imagenet_val_lmdb,
don't use val.txt but instead some val_nonexistant_file.txt
Observe that none of the tools warn that this may not do anything useful.

```
I0510 21:48:31.669347 16480 db_lmdb.cpp:35] Opened lmdb caffe_data/imagenet_val_lmdb
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1462909711 (unix time) try "date -d @1462909711" if you are using GNU date ***
PC: @     0x7f6cc1c16478 gsignal
*** SIGABRT (@0x3e800004059) received by PID 16473 (TID 0x7f6c90a0b700) from PID 16473; stack trace: ***
    @     0x7f6cc1f9bd30 (unknown)
    @     0x7f6cc1c16478 gsignal
    @     0x7f6cc1c178fa abort
    @     0x7f6cc47fa32d __gnu_cxx::__verbose_terminate_handler()
    @     0x7f6cc47f83b6 (unknown)
    @     0x7f6cc47f8401 std::terminate()
    @     0x7f6cc47f8619 __cxa_throw
    @     0x7f6cc47f8b2c operator new()
    @     0x7f6cc72af8b7 std::__cxx11::basic_string<>::_M_construct<>()
    @     0x7f6cc72b06eb _ZN5caffe2db10LMDBCursor5valueB5cxx11Ev
    @     0x7f6cc72c3a7e caffe::DataReader::Body::read_one()
    @     0x7f6cc72c3ea7 caffe::DataReader::Body::InternalThreadEntry()
    @     0x7f6cc4afd105 (unknown)
    @     0x7f6cc1f92454 start_thread
    @     0x7f6cc1ccbeed clone
    @                0x0 (unknown)
Abgebrochen

$ echo '_ZN5caffe2db10LMDBCursor5valueB5cxx11Ev' | c++filt
caffe::db::LMDBCursor::value[abi:cxx11]()
```

This is always reproducible.

Here's what exactly I did, from clone to crash,
and what valgrind says about it: http://pastebin.com/rw4zCaph

Note that I'm only submitting this for:
- increased robustness of caffe, because a warning/explanation is nicer than a stackdump
- to give (for now) others a chance to find out why caffe is crashing

I'm NOT submitting this to get help about this issue. (If you're reading this because you have the same issue: duh, don't try to use an empty lmdb, that's just silly!)

This issue doesn't seem to be known by the issue tracker or the mailing list.
This is not a request for help. I can understand if someone doesn't see this as a bug report.
Therefore, I'll claim this to be an "enhancement".

**Comment by arjun-kava** [comment]
same here

**Comment by AndrewZhou924** [comment]
Hi buddy, do you solve this issue?

**Comment by RuaYahya** [comment]
if any one find a soultion , please advice us.
